Branch-SDK/src/io/branch/referral/indexing/BranchUniversalObject.java

```diff
@@ -0,0 +1,506 @@
+                    branchUniversalObject = new BranchUniversalObject();
+
+                    if (latestParam.has(Defines.Jsonkey.ContentTitle.getKey())) {
+                        branchUniversalObject.title_ = latestParam.getString(Defines.Jsonkey.ContentTitle.getKey());
+                        latestParam.remove(Defines.Jsonkey.ContentTitle.getKey());
+                    }
+                    if (latestParam.has(Defines.Jsonkey.CanonicalIdentifier.getKey())) {
+                        branchUniversalObject.canonicalIdentifier_ = latestParam.getString(Defines.Jsonkey.CanonicalIdentifier.getKey());
+                        latestParam.remove(Defines.Jsonkey.CanonicalIdentifier.getKey());
+                    }
+                    if (latestParam.has(Defines.Jsonkey.ContentKeyWords.getKey())) {
+                        JSONArray keywordJsonArray = latestParam.getJSONArray(Defines.Jsonkey.ContentKeyWords.getKey());
+                        for (int i = 0; i < keywordJsonArray.length(); i++) {
+                            branchUniversalObject.keyWords_.add((String) keywordJsonArray.get(i));
+                        }
+                        latestParam.remove(Defines.Jsonkey.ContentKeyWords.getKey());
+                    }
+                    if (latestParam.has(Defines.Jsonkey.ContentDesc.getKey())) {
+                        branchUniversalObject.description_ = latestParam.getString(Defines.Jsonkey.ContentDesc.getKey());
+                        latestParam.remove(Defines.Jsonkey.ContentDesc.getKey());
+                    }
+                    if (latestParam.has(Defines.Jsonkey.ContentImgUrl.getKey())) {
+                        branchUniversalObject.imageUrl_ = latestParam.getString(Defines.Jsonkey.ContentImgUrl.getKey());
+                        latestParam.remove(Defines.Jsonkey.ContentImgUrl.getKey());
+                    }
+                    if (latestParam.has(Defines.Jsonkey.ContentType.getKey())) {
+                        branchUniversalObject.type_ = latestParam.getString(Defines.Jsonkey.ContentType.getKey());
+                        latestParam.remove(Defines.Jsonkey.ContentType.getKey());
+                    }
+                    if (latestParam.has(Defines.Jsonkey.ContentExpiryTime.getKey())) {
+                        branchUniversalObject.expiration_ = latestParam.getLong(Defines.Jsonkey.ContentExpiryTime.getKey());
+                        latestParam.remove(Defines.Jsonkey.ContentExpiryTime.getKey());
+                    }
+
+                    ///-----------Link Properties----------------///
+                    if (latestParam.has("~channel")) {
+                        branchUniversalObject.linkProperties_.setChannel(latestParam.getString("~channel"));
+                        latestParam.remove("~channel");
+                    }
+                    if (latestParam.has("~creation_source")) {
+                        latestParam.remove("~creation_source");
```

Comment:
User has specified Tags,Feature etc in Link params. we are recreating the object with LinkParams
Do all the fields are in there as they specified.  For example they can get tags by
`branchUniversalObject.getLinkProperties().getTags()`